expo/expo
# [docs] Add redesigned button components

# Why

Part of [ENG-3962](https://linear.app/expo/issue/ENG-3962/add-new-components-as-separate-prs-in-expoexpo)

This adds the button component(s) to the docs.

light | dark
--- | ---
![image](https://user-images.githubusercontent.com/1203991/153761783-7740efc8-a6a3-42cb-b4fe-3aa17472fde7.png) | ![image](https://user-images.githubusercontent.com/1203991/153761789-f76975cc-864d-412b-af3b-521c7fede96e.png)

<details><summary>Code of used test page</summary>

```jsx
---
title: Test Buttons
hideTOC: true
---

import { Button } from '~/ui/components/Button';
import { ArrowLeftIcon, ArrowRightIcon } from '~/ui/foundations/icons';

theme | mini | small | large
--- | --- | --- | ---
primary _(default)_ | <Button size="mini">Primary</Button> | <Button size="small">Primary</Button> | <Button size="large">Primary</Button>
secondary | <Button size="mini" theme="secondary">Secondary</Button> | <Button size="small" theme="secondary">Secondary</Button> | <Button size="large" theme="secondary">Secondary</Button>
tertiary | <Button size="mini" theme="tertiary">Tertiary</Button> | <Button size="small" theme="tertiary">Tertiary</Button> | <Button size="large" theme="tertiary">Tertiary</Button>
transparent | <Button size="mini" theme="transparent">Transparent</Button> | <Button size="small" theme="transparent">Transparent</Button> | <Button size="large" theme="transparent">Transparent</Button>
ghost | <Button size="mini" theme="ghost">Ghost</Button> | <Button size="small" theme="ghost">Ghost</Button> | <Button size="large" theme="ghost">Ghost</Button>

**Ghost buttons with icons**

<div style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between', margin: '2.5ch 0' }}>
  <Button theme="ghost" icon={<ArrowLeftIcon />} href="#home">Home</Button>
  <Button theme="ghost" iconRight={<ArrowRightIcon />} href="#create-a-new-app">Create a new app</Button>
</div>
```

</details>

# How

- Copied the button components from website
- Removed form state for now

# Test Plan

- Copy test code
- Check if buttons are working fine

# Checklist
